pkg/segment/writer/segstore.go

```diff
@@ -1513,13 +1513,16 @@ func writeSstToBuf(sst *structs.SegStats, buf []byte) (uint16, error) {
 	idx += 2
 
 	// HLL_Data
-	hllByteSlice := sst.GetHllBytesInPlace(buf[idx : idx+uint16(hllDataSize)])
+	hllDataSliceFullCap := buf[idx : idx+uint16(hllDataSize)]
+	hllByteSlice := hllDataSliceFullCap[:len(hllDataSliceFullCap):len(hllDataSliceFullCap)] // len = cap
+	hllByteSlice = sst.GetHllBytesInPlace(hllByteSlice)
```

Comment:
This is interesting. I'd recommend adding a bit longer of a comment explaining why it's necessary that we go through these extra steps to create a buffer where the length and capacity are the same, since we generally don't need to do that